src/models/UserModel.js

```diff
@@ -0,0 +1,43 @@
+import {mapApiResponse} from "@/helpers/objectModification";
+import {format} from "date-fns";
+
+export class User{
+    constructor(jsonResponse) {
+        const mappedResponse = mapApiResponse(jsonResponse);
+        this.id= mappedResponse.id;
+        this.username= mappedResponse.username;
+        this.firstName= mappedResponse.firstName;
+        this.lastName= mappedResponse.lastName;
+        this.language= mappedResponse.language;
+        this.dsgvoAccepted= mappedResponse.dsgvoAccepted;
+        this.dateJoined= new Date(mappedResponse.dateJoined);
+        this.modifiedAt= new Date(mappedResponse.modifiedAt);
+        this.lastLogin= new Date(mappedResponse.lastLogin);
+        this.isSuperuser= mappedResponse.isSuperuser;
+        this.onboardingPassed= mappedResponse.onboardingPassed;
+    }
+    #toPayload() {
+        // Method is considered private since
+        // No usage planned for now, but in later releases.
```

Comment:
Wäre die usage nicht `dsgvo_accepted`, `onboarding_passed` oder auch `language` zu speichern? Oder können wir das über den store besser?

Reply:
Ich dachte an Methoden die ausschließlich  dsgvo_accepted/onboarding_passed setzen und eine die ausschließlich setzt.

Die `toPayload` methoden sind dafür gedacht wenn man durch Userinput, bspw. in einer Form, mehrere Felder speichert.